### DISCUSSION TITLE
Saving additional information on import

### DISCUSSION THREAD

**Discussion by JOJ0** [beetbox/beets#4212] [Q&A] [answered]
I'd like to achieve the following on import:

Keep record of the path of the original file was residing in before moving into beets folder structure (`import: copy: yes` is set)

The reason why I'd like to keep this is that it contains useful information about the style of music my flatmates and me sorted this music in back in the day when this mp3 library was collected and used.

I now want to import / retag / file everything with beets once and for all but keep this valuable information in a db fileld, because I might want to refer to it in future DJ use of this library.

I already found the option

`--set=FIELD=VALUE  -- set the given fields to the supplied values`

of the importer but I am not sure if I can use it to automatically find out the original folder name and save in this db field. Actually I only need parts of the original folder name and would would be nice to cut away things before importing into this filed.

Thanks in advance!

**Comment by JOJ0** [comment, marked as answer]
I now just do that manually, using --set.
Another option would be to use this plugin called dirfields: https://github.com/mried/beetsplug/wiki/dirfields

**Reply by RollingStar** [reply]
Funny, I was working on something like this. Little did I know someone resolved it half a decade ago.

I still think saving the path should be default behavior - little harm done if kept, but advanced users may want to disable it.

https://discourse.beets.io/t/should-we-start-recording-the-original-path-and-filename-on-import/1494

Not directly applicable, but freezetag is notable also. https://github.com/x1ppy/freezetag

**Comment by JOJ0** [comment]
I talked to the original dev, forked and updated the dirfields plugin and will continue maintenance, also it has got an installer now: https://github.com/JOJ0/beets-dirfields 🎉
